gufe/components/proteincomponent.py

```diff
@@ -315,39 +232,29 @@ def _from_dict(cls, ser_dict: dict, name: str = ""):
         # Add Atoms
         for atom in ser_dict["atoms"]:
             atomic_num = int(atom[0])
-            atomic_name = atom[1]
-            atomic_fc = atom[2]
-            atomic_arom = eval(str(atom[3]))
-            atomic_ste = atom[4]
-            atomic_props = atom[5]
-            atom_mi_dict = atom[6]
 
             a = Atom(atomic_num)
-            a.SetAtomMapNum(atomic_props["id"])
-            a.SetFormalCharge(atomic_fc)
-            a.SetIsAromatic(atomic_arom)
-            # a.SetChiralTag(atomic_ste)
-
-            # put mi_dict back to class
-            atom_monomerInfo = Chem.AtomPDBResidueInfo()
-            for key, val in atom_mi_dict.items():
-                f = getattr(atom_monomerInfo, "Set" + str(key))
-                f(val)
+            mi = Chem.AtomPDBResidueInfo()
 
-            a.SetMonomerInfo(atom_monomerInfo)
+            mi.SetChainId(atom[1])
+            mi.SetSerialNumber(atom[2])
+            mi.SetSegmentNumber(atom[3])
+            mi.SetInsertionCode(atom[4])
+            mi.SetName(atom[5])
+            mi.SetResidueName(atom[6])
+            mi.SetResidueNumber(int(atom[7]))
+            mi.SetIsHeteroAtom(atom[8] == 'Y')
+            a.SetFormalCharge(atom[9])
 
-            for prop_name, prop_value in atomic_props.items():
-                assign_correct_prop_type(
-                    rd_obj=a, prop_name=prop_name, prop_value=prop_value
-                )
+            a.SetMonomerInfo(mi)
```

Comment:
Brilliant! Happy to see us shift from using properties to inbuilt pieces of RDKit mols.